test/cets_SUITE.erl

```diff
@@ -2473,15 +2473,38 @@ disco_node_start_timestamp_is_updated_after_node_restarts(Config) ->
     simulate_disco_restart(Setup),
     wait_for_disco_timestamp_to_be_updated(Disco, node_start_timestamps, Node2, OldTimestamp).
 
-disco_pang_result_arrives_after_nodeup(Config) ->
+disco_late_pang_result_arrives_after_node_went_up(Config) ->
+    Node1 = node(),
+    #{ct2 := Node2} = proplists:get_value(nodes, Config),
+    %% unavailable_nodes list contains nodes which have not responded to pings.
+    %% Ping is async though.
+    %% So, there could be the situation when the result of ping would be processed
+    %% after the node actually got connected.
     meck:new(cets_ping, [passthrough]),
+    Me = self(),
     meck:expect(cets_ping, send_ping_result, fun(Pid, Node, _PingResult) ->
+        %% Wait until Node is up
+        Cond = fun() -> lists:member(Node, nodes()) end,
+        cets_test_wait:wait_until(Cond, true),
+        Me ! send_ping_result_called,
+        %% Similate pang returned to cets_discovery.
+        %% cets_join does not use send_ping_result function
+        %% and would receive pong and join correctly.
         meck:passthrough([Pid, Node, pang])
     end),
     try
-        Setup = setup_two_nodes_and_discovery(Config, [wait]),
+        %% setup_two_nodes_and_discovery would call disconnect_node node
+        %% before connecting it back.
+        Setup = setup_two_nodes_and_discovery(Config, [wait, disco2]),
         #{disco_name := DiscoName} = Setup,
-        ?assertMatch([], maps:get(unavailable_nodes, cets_status:status(DiscoName)))
+        Status = cets_status:status(DiscoName),
+        %% Check that pang is ignored and unavailable_nodes list is empty.
+        ?assertMatch([], maps:get(unavailable_nodes, Status)),
+        %% Ensure that Node2 is in a list of joined nodes
+        ?assertMatch([Node1, Node2], maps:get(joined_nodes, Status)),
+        %% Ensure that send_ping_result was called
+        %% (it is not called if node is in the nodes() list)
+        receive_message(send_ping_result_called)
```

Comment:
I doesn't make sense to check it after the testcase, because it might have arrived after we checked unavailable nodes.

Reply:
it makes some sense, because here is how the function looks like:
```
ping_not_connected_nodes(Nodes) ->
    Self = self(),
    NotConNodes = Nodes -- [node() | nodes()],
    [
        spawn_link(fun() -> cets_ping:send_ping_result(Self, Node, cets_ping:ping(Node)) end)
     || Node <- lists:sort(NotConNodes)
    ],
    ok.
 ```
 
 So there is a chance that node got connected by the reason not related to the current test - in this case `cets_ping:send_ping_result/2` would never be called :)